debug/src/debug.js

```diff
@@ -83,6 +84,32 @@ export function initDebug() {
 
 		if (oldBeforeDiff) oldBeforeDiff(vnode);
 	};
+
+	options.diffed = (vnode) => {
+		if (vnode._component && vnode._component.__hooks && vnode._component.__hooks._pendingEffects.length > 0) {
+			vnode._component.__hooks._pendingEffects.forEach((x) => {
```

Comment:
Maybe we can simplify this check and combine it with the one a few lines down:

```js
if (vnode._component && vnode._component._hooks) {
  let hooks = vnode._component._hooks;

  if (hooks._pendingEffects.length > 0) {
    // ...
  }
  else if (hooks._pendingLayoutEffects.length > 0) {
    // ...
  }
}

```

Reply:
That's a good question. Personally I'd add it. If there really is a valid reason to trigger `useEffect` on each render one could just pass an always changing dependency:

```jsx
let i = 0;
useEffect(() => x, [i++]);
```